src/ba-transport.c

```diff
@@ -267,8 +276,15 @@ int ba_transport_thread_set_state(
 
 	pthread_mutex_unlock(&th->state_mtx);
 
-	if (!skip)
+	if (!skip) {
 		pthread_cond_signal(&th->changed);
+		if (state != old_state) {
+			if (old_state == BA_TRANSPORT_THREAD_STATE_RUNNING ||
+					state == BA_TRANSPORT_THREAD_STATE_RUNNING) {
+				eventfd_write(th->state_fd, 1);
```

Comment:
Why you are starting IO watch for property change notification? I think that it will be easier to call `bluealsa_dbus_pcm_update()` directly from here (with some "ifs" for checking which PCM is associated with the transport thread).

Reply:
I wasn't sure about libdus thread management internals, so I wanted this dbus blockng call to be made from the glib mainloop thread. However, if if is safe to call this from transport IO threads then that will simplify things a lot and I will make that change.

Reply:
Yes, it should be safe to call it from here, GIO is thread safe (exception from general rule in glib, that glib APIs are not thread-safe).

And one more thing. If this PR is supposed to be tested by someone else, I think that it will be better to base in on 1326cb1 (not on master). After commit c0b1d29, there is a rarely occurring dead-lock when opening PCM (added mutex when accessing pcm->fd in io.c). Right now I'm working on improvements for thread/mutex management and more patches are coming (it's quite possible that I will abandon pthraed cancellation.... - it seems that there is a bug in libc: dead-lock in the scenario like this: cancel signal -> running some cancellation internals (lock mutex) -> another signal (e.g. SIGTERM) -> signal handler runs libc write -> running [again] cancellation internals -> dead lock on mutex lock). If testing this PR is not critical, you can wait for a proper fix in master (I will push it in few days) - the fix will add dedicated mutex for pcm client in the `bluealsa_pcm_open()`, but I'm still testing it though.

Reply:
OK, I will remove the state_fd eventfd and update the D-Bus state from the IO threads. As far as testing goes, I have enough feedback already from my own tests and #536 to be confident that is is a worth while addition, so I will stop trying to keep it based on master and wait for your current re-design to be completed before re-basing.

Reply:
I've just pushed the commit with dedicated client mutex lock. I hope that CI will not fail so often any more (it still can fail because of mentioned libc signal issue, but it's manifesting itself only during bluealsa termination with SIGTERM/SIGINT).

Reply:
Hmm... I have not thought about that, but yes, you might be right. I will have to check that. Thx!

Reply:
With your suggestion (masking all signals) I've been running `test_playback_hw_set_free` test in a loop for ~12 hours and no deadlock has accrued. Without masking, I've got deadlock within ~30 minutes (running on RPi4 Debian 11.5). Once again thanks for suggestion!